src/fidesops/models/fidesops_user.py

```diff
@@ -48,3 +57,18 @@ def credentials_valid(self, password: str) -> bool:
         )
 
         return provided_password_hash == self.hashed_password
+
+    def login(self, db: Session) -> None:
+        """Performs any update to the user model object needed when logging in"""
+        self.last_login_at = datetime.utcnow()
+        self.save(db)
+
+    def update_password(self, db: Session, new_password: str) -> None:
+        """Updates the user's password to the specified value.
+        No validations are performed on the old/existing password within this function."""
+
+        hashed_password, salt = FidesopsUser.hash_password(new_password)
+        self.hashed_password = hashed_password
+        self.salt = salt
+        self.password_reset_at = datetime.utcnow()
+        self.save(db)
```

Comment:
It'd be nice to have a model-level test on this method - nice job moving the hash_password logic into its own class method to be shared between create/update.

Reply:
agreed - will do.

Reply:
added in https://github.com/ethyca/fidesops/pull/394/commits/0c6766548f54a58e1a27f5c41bd49a9b8eacdc62